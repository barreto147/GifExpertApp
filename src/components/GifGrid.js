//import { useState, useEffect } from "react";
//import { getGifs } from "../helpers/getGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      <div className="card-grid animate__animated animate__zoomInUp ">
        {loading && <p>Loading</p>}
        {images.map((image) => {
          return <GifGridItem key={image.id} {...image} />;
        })}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
