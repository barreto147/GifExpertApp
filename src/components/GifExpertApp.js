import { useState } from "react";
import { CategoryAdd } from "./CategoryAdd";
import { GifGrid } from "./GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  //   const handleAdd = () => {
  //     //setCategories([...categories, "Digimon"]);
  //     setCategories((categ) => [...categ, "Digimon"]);
  //   };

  return (
    <>
      <h2>GifExpertApp</h2>
      <CategoryAdd setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
