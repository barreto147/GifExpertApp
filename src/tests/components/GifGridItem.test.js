import { GifGridItem } from "../../components/GifGridItem";
import { shallow } from "enzyme";

describe("Pruebas de GifGridItem", () => {
  const title = "Un titulo";
  const url = "https://localhost/algo.png";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Prueba de Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el titulo", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("La imagen debe tener una url en el src y alt en los props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("el div debe tener la clase de animate__fadeIn", () => {
    const div = wrapper.find("div");
    const animateClass = div.prop("className");

    expect(animateClass.includes("animate__fadeIn")).toBe(true);
  });
});
