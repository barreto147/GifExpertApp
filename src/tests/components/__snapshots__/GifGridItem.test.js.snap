// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGridItem Prueba de Snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.png"
  />
  <p>
    Un titulo
  </p>
</div>
`;
